tests/Unit/Campaigns/Routes/CampaignOverviewStatisticsTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Give\Tests\Unit\Campaigns\Routes;
+
+use DateTime;
+use Give\Campaigns\Models\Campaign;
+use Give\Campaigns\Routes\CampaignOverviewStatistics;
+use Give\DonationForms\Models\DonationForm;
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationStatus;
+use Give\Framework\Database\DB;
+use Give\Framework\Support\ValueObjects\Money;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+use WP_REST_Request;
+
+/**
+ * @unreleased
+ */
+final class CampaignOverviewStatisticsTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * @unreleased
+     */
+    public function testReturnsAllTimeDonationsStatistics()
+    {
+        $campaign = Campaign::factory()->create();
+        $form = DonationForm::factory()->create();
+
+        $db = DB::table('give_campaign_forms');
+        $db->insert(['form_id' => $form->id, 'campaign_id' => $campaign->id]);
+
+        $donation1 = Donation::factory()->create([
+            'formId' => $form->id,
+            'status' => DonationStatus::COMPLETE(),
+            'amount' => new Money(1000, 'USD'),
+            'createdAt' => new DateTime('-35 days'),
+        ]);
+        give_update_meta($donation1->id, '_give_completed_date', $donation1->createdAt->format('Y-m-d H:i:s'));
+
+        $donation2 = Donation::factory()->create([
+            'formId' => $form->id,
+            'status' => DonationStatus::COMPLETE(),
+            'amount' => new Money(1000, 'USD'),
+            'createdAt' => new DateTime('-5 days'),
+        ]);
+        give_update_meta($donation2->id, '_give_completed_date', $donation2->createdAt->format('Y-m-d H:i:s'));
+
+        $donation3 = Donation::factory()->create([
+            'formId' => $form->id,
+            'status' => DonationStatus::COMPLETE(),
+            'amount' => new Money(1000, 'USD'),
+            'createdAt' => new DateTime('now'),
+        ]);
+        give_update_meta($donation3->id, '_give_completed_date', $donation3->createdAt->format('Y-m-d H:i:s'));
+
+        $request = new WP_REST_Request('GET', '/give-api/v2/campaign-overview-statistics');
+        $request->set_param('campaignId', $campaign->id);
+
+        $route = new CampaignOverviewStatistics;
+        $response = $route->handleRequest($request);
+
+        $this->assertEquals(3, $response[0]['donorCount']);
+        $this->assertEquals(3, $response[0]['donationCount']);
+        $this->assertEquals(30, $response[0]['amountRaised']);
```

Comment:
We need to update these lines to use `$response->data[INDEX]` after changing the `handleRequest` return type to `WP_REST_Response`

Reply:
Resolved, see [4b45c15](https://github.com/impress-org/givewp/pull/7545/commits/4b45c15909c3d10e81d395d37152bb2816ef35a5).